### DISCUSSION TITLE
Open an Extension on landing page.

### DISCUSSION THREAD

**Discussion by kumbham** [eclipse-theia/theia#9690] [General] [answered]
How do I open an extension screen when I load theia? `http://localhost:3000` opens the theia with file explorer as the default screen. Is there a way I could pass additional metadata in the URL to open an extension UI instead of file explorer?
for example, I want to open SCM extension on loading theia instead of file explorer. Any help would be appreciated, thanks!

![Screen Shot 2021-07-01 at 10 05 33 AM](https://user-images.githubusercontent.com/9061872/124163311-54d0a500-da54-11eb-90e0-3c79b8114bc1.png)

**Comment by vince-fugnitto** [comment, marked as answer]
@kumbham thank you for the discussion, you'll need to implement a [custom theia extension](https://theia-ide.org/docs/authoring_extensions/) which modifies the default behavior, and activates or reveals the specific widget or view you want on startup.

You can take a look at the `@theia/getting-started` extension as an example since it opens the `getting-started` page on startup.
In your case, you can implement it for whichever view you want to open on startup, for whichever condition satisfies your use-case. The `getting-started` view opens by default when no workspace is present:

https://github.com/eclipse-theia/theia/blob/9648953cd6e8552b7e027558509ba77770bd8b6f/packages/getting-started/src/browser/getting-started-contribution.ts#L51-L57

It uses the `onStart` lifecycle method: https://eclipse-theia.github.io/theia/docs/next/interfaces/core.frontendapplicationcontribution-1.html#onstart.
